pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/SetOperator.java

```diff
@@ -89,12 +92,23 @@ public ExecutionStatistics getExecutionStatistics() {
 
   @Override
   protected TransferableBlock getNextBlock() {
-    // A blocking call to construct a set with all the right side rows.
-    if (!_isRightSetBuilt) {
-      constructRightBlockSet();
+    try {
+      if (_isTerminated) {
+        return TransferableBlockUtils.getEndOfStreamTransferableBlock();
+      }
+      if (!_isRightSetBuilt) {
+        // construct a SET with all the right side rows.
+        constructRightBlockSet();
+      }
+      if (_upstreamErrorBlock != null) {
```

Comment:
We probably don't need to save this as member variable. Also, the error from right table is not gathered